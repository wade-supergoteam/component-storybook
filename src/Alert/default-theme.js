// @flow

export const primary = "#007bff";
export const secondary = "#6c757d";
export const success = "#28a745";
export const info = "#17a2b8";
export const warning = "#ffc107";
export const danger = "#dc3545";
export const light = "#f8f9fa";
export const dark = "#343a40";

export const alertPaddingY = ".75rem";
export const alertPaddingX = "1.25rem";
export const alertMarginBottom = "1rem";
export const alertBorderRadius = "0.25rem";
export const alertLinkFontWeight = "bold";
export const alertBorderWidth = "1px";

export const alertBgLevel = -10;
export const alertBorderLevel = -9;
export const alertColorLevel = 6;
