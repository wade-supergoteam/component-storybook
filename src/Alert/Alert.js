// @flow

// $FlowIssue
import get from "lodash.get";
import PropTypes from "prop-types";
import styled from "styled-components";
import { borderRadius } from "../utils/border-radius";
import { themeColorLevel } from "../utils/color-functions";
import themeProp from "../utils/theme";
import * as defaultTheme from "./default-theme";
import { alertVariant } from "./utils/alert-variant";

const {
  alertPaddingY,
  alertPaddingX,
  alertMarginBottom,
  alertBorderWidth,
  alertLinkFontWeight,
  alertBorderRadius,
  alertBgLevel,
  alertBorderLevel,
  alertColorLevel,
} = defaultTheme;

const Alert = styled.div`
  position: relative;
  padding: ${themeProp("alertPaddingY", alertPaddingY)} ${themeProp("alertPaddingX,", alertPaddingX)};

  margin-bottom: ${themeProp("alertMarginBottom", alertMarginBottom)};

  border: ${themeProp("alertBorderWidth", alertBorderWidth)} solid transparent;

  a {
    font-weight: ${themeProp("alertLinkFontWeight", alertLinkFontWeight)};
  }

  ${borderRadius(themeProp("alertBorderRadius", alertBorderRadius))};

  ${({ type, theme }) =>
    alertVariant(
      themeColorLevel(defaultTheme[type], get(theme, "alertBgLevel", alertBgLevel)),
      themeColorLevel(defaultTheme[type], get(theme, "alertBorderLevel", alertBorderLevel)),
      themeColorLevel(defaultTheme[type], get(theme, "alertColorLevel", alertColorLevel)),
    )}};
`;

Alert.defaultProps = {
  type: "primary",
};

Alert.propTypes = {
  type: PropTypes.oneOf(["primary", "secondary", "success", "danger", "warning", "info", "light", "dark"]),
};

export default Alert;
