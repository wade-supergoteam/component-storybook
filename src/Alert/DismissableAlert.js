import React, { cloneElement, isValidElement } from "react";
import PropTypes from "prop-types";
import omit from "lodash.omit";
import styled, { css } from "styled-components";
import themeProp from "../utils/theme";
import { hoverFocus } from "../utils/hover";

import Alert from "./Alert";
import CloseIcon from "../CloseIcon";

import { alertPaddingY, alertPaddingX } from "./default-theme";

const closeIconStyles = css`
  position: absolute;
  top: 0;
  right: 0;
`;

export const StyledCloseIcon = styled(CloseIcon)`
  padding: ${themeProp("alertPaddingY", alertPaddingY)} ${themeProp("alertPaddingX", alertPaddingX)};
  color: ${props => !props.theme.closeColor && "inherit"};
  ${closeIconStyles};
`;

const CloseString = styled.button`
  ${closeIconStyles};
  padding: ${themeProp("alertPaddingY", alertPaddingY)} ${themeProp("alertPaddingX", alertPaddingX)};
  cursor: pointer;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
  opacity: 0.5;

  ${hoverFocus(`
    color: #000;
    text-decoration: none;
    opacity: .75;
  `)};
`;

const DismissableAlert = props => {
  const filteredProps = omit(props, ["onClose", "closeIcon"]);
  return (
    <Alert {...filteredProps}>
      {props.children}

      {props.closeIcon && isValidElement(props.closeIcon) ? (
        cloneElement(props.closeIcon, { onClick: props.onClose })
      ) : (
        <CloseString onClick={props.onClose}>{props.closeIcon}</CloseString>
      )}

      {!props.closeIcon && <StyledCloseIcon onClick={props.onClose} />}
    </Alert>
  );
};

DismissableAlert.propTypes = {
  closeIcon: PropTypes.oneOf([PropTypes.string, PropTypes.node]),
};

DismissableAlert.defaultProps = {
  closeIcon: null,
};

export default DismissableAlert;
