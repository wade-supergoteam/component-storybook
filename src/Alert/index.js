import DismissableAlert, { StyledCloseIcon as CloseIcon } from "./DismissableAlert";

export { alertVariant } from "./utils/alert-variant";

export { DismissableAlert, CloseIcon };

export { default } from "./Alert";
