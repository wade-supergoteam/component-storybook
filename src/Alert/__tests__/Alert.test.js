import React from "react";
import renderer from "react-test-renderer";
import DismissableAlert from "../DismissableAlert";
import Alert from "..";
import "jest-styled-components";

test("Create Alert", () => {
  const wrapper = renderer.create(<Alert />);
  expect(wrapper).toMatchSnapshot();
});

test("Create DismissableAlert", () => {
  const wrapper = renderer.create(<DismissableAlert type="primary" />);
  expect(wrapper).toMatchSnapshot();
});

test("Create DismissableAlert with onClose", () => {
  const wrapper = renderer.create(<DismissableAlert type="primary" onClose={() => this.closeAlert("primary")} />);
  expect(wrapper).toMatchSnapshot();
});
