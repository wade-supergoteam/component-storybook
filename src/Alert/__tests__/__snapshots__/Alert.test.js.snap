// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Alert 1`] = `
.c0 {
  position: relative;
  padding: .75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #08488c;
  background-color: #cce4ff;
  border-color: #b7daff;
}

.c0 a {
  font-weight: bold;
}

.c0 hr {
  border-top-color: #9ecdff;
}

.c0 a {
  color: #052f5c;
}

<div
  className="c0"
  type="primary"
/>
`;

exports[`Create DismissableAlert 1`] = `
.c0 {
  position: relative;
  padding: .75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #08488c;
  background-color: #cce4ff;
  border-color: #b7daff;
}

.c0 a {
  font-weight: bold;
}

.c0 hr {
  border-top-color: #9ecdff;
}

.c0 a {
  color: #052f5c;
}

.c3 {
  font-size: 1.5rem;
  font-weight: 700;
  line-height: 1;
  color: #000;
  text-shadow: 0 1px 0 #fff;
  opacity: 0.5;
  padding: 0;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
}

.c3:focus,
.c3:hover {
  color: (props = {}) =>(0,_lodash2.default)(props.theme,path,defaultValue);
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: .75;
}

.c3:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.c2 {
  padding: .75rem 1.25rem;
  color: inherit;
  position: absolute;
  top: 0;
  right: 0;
}

.c1 {
  position: absolute;
  top: 0;
  right: 0;
  padding: .75rem 1.25rem;
  cursor: pointer;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
  opacity: 0.5;
}

.c1:focus,
.c1:hover {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: .75;
}

<div
  className="c0"
  type="primary"
>
  <button
    className="c1"
    onClick={undefined}
  />
  <button
    aria-label="Close"
    className="c2 c3"
    onClick={undefined}
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;

exports[`Create DismissableAlert with onClose 1`] = `
.c0 {
  position: relative;
  padding: .75rem 1.25rem;
  margin-bottom: 1rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #08488c;
  background-color: #cce4ff;
  border-color: #b7daff;
}

.c0 a {
  font-weight: bold;
}

.c0 hr {
  border-top-color: #9ecdff;
}

.c0 a {
  color: #052f5c;
}

.c3 {
  font-size: 1.5rem;
  font-weight: 700;
  line-height: 1;
  color: #000;
  text-shadow: 0 1px 0 #fff;
  opacity: 0.5;
  padding: 0;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
}

.c3:focus,
.c3:hover {
  color: (props = {}) =>(0,_lodash2.default)(props.theme,path,defaultValue);
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: .75;
}

.c3:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.c2 {
  padding: .75rem 1.25rem;
  color: inherit;
  position: absolute;
  top: 0;
  right: 0;
}

.c1 {
  position: absolute;
  top: 0;
  right: 0;
  padding: .75rem 1.25rem;
  cursor: pointer;
  background: transparent;
  border: 0;
  -webkit-appearance: none;
  opacity: 0.5;
}

.c1:focus,
.c1:hover {
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
  opacity: .75;
}

<div
  className="c0"
  type="primary"
>
  <button
    className="c1"
    onClick={[Function]}
  />
  <button
    aria-label="Close"
    className="c2 c3"
    onClick={[Function]}
  >
    <span
      aria-hidden="true"
    >
      ×
    </span>
  </button>
</div>
`;
